core/src/main/java/com/linecorp/armeria/common/HttpData.java

```diff
@@ -83,7 +82,11 @@ static HttpData wrap(byte[] data, int offset, int length) {
             return EMPTY_DATA;
         }
 
-        return new ByteBufHttpData(Unpooled.wrappedBuffer(data, offset, length), false);
+        if (data.length == length) {
```

Comment:
`offset == 0 &&` ?

Reply:
I don't think it's possible because of the index check above

Reply:
Ah yeah. :sweat_smile: